File: flutter/test/state/capture_status_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:theta_client_flutter/theta_client_flutter.dart';

void main() {
  setUp(() {});

  tearDown(() {});

  test('CaptureStatusEnum', () async {
    List<List<dynamic>> data = [
      [CaptureStatusEnum.unknown, 'UNKNOWN'],
      [CaptureStatusEnum.shooting, 'SHOOTING'],
      [CaptureStatusEnum.idle, 'IDLE'],
      [CaptureStatusEnum.selfTimerCountdown, 'SELF_TIMER_COUNTDOWN'],
      [CaptureStatusEnum.bracketShooting, 'BRACKET_SHOOTING'],
      [CaptureStatusEnum.converting, 'CONVERTING'],
      [CaptureStatusEnum.timeShiftShooting, 'TIME_SHIFT_SHOOTING'],
      [CaptureStatusEnum.continuousShooting, 'CONTINUOUS_SHOOTING'],
      [
        CaptureStatusEnum.retrospectiveImageRecording,
        'RETROSPECTIVE_IMAGE_RECORDING'
      ],
      [CaptureStatusEnum.burstShooting, 'BURST_SHOOTING'],
    ];
    expect(data.length, CaptureStatusEnum.values.length, reason: 'enum count');
    for (int i = 0; i < data.length; i++) {
      expect(data[i][0].toString(), data[i][1], reason: data[i][1]);
    }
  });
}

File: react-native/src/__tests__/theta-repository/theta-state/capture-status.test.ts
import { CaptureStatusEnum } from '../../../theta-repository/theta-state';

describe('CaptureStatusEnum', () => {
  const data: [CaptureStatusEnum, string][] = [
    [CaptureStatusEnum.UNKNOWN, 'UNKNOWN'],
    [CaptureStatusEnum.SHOOTING, 'SHOOTING'],
    [CaptureStatusEnum.IDLE, 'IDLE'],
    [CaptureStatusEnum.SELF_TIMER_COUNTDOWN, 'SELF_TIMER_COUNTDOWN'],
    [CaptureStatusEnum.BRACKET_SHOOTING, 'BRACKET_SHOOTING'],
    [CaptureStatusEnum.CONVERTING, 'CONVERTING'],
    [CaptureStatusEnum.TIME_SHIFT_SHOOTING, 'TIME_SHIFT_SHOOTING'],
    [CaptureStatusEnum.CONTINUOUS_SHOOTING, 'CONTINUOUS_SHOOTING'],
    [
      CaptureStatusEnum.RETROSPECTIVE_IMAGE_RECORDING,
      'RETROSPECTIVE_IMAGE_RECORDING',
    ],
    [CaptureStatusEnum.BURST_SHOOTING, 'BURST_SHOOTING'],
  ];

  test('length', () => {
    expect(data.length).toBe(Object.keys(CaptureStatusEnum).length);
  });

  test('data', () => {
    data.forEach((item) => {
      expect(item[0]).toBe(item[1]);
    });
  });
});
